system/datatypes/peer_reviewed_article.php

```diff
@@ -576,262 +239,235 @@ public function indexAction() {
+		$form->updateDom();
+
+		/** @var HyphaDomElement $main */
+		$main = $this->html->find('#main');
+		$main->append($form);
+
+		/** @var HyphaDomElement $commands */
+		$commands = $this->html->find('#pageCommands');
+		$commands->append($this->makeActionButton(__('art-cancel')));
+		$commands->append($this->makeActionButton(__('art-save'), self::PATH_EDIT, self::CMD_SAVE));
+		/** @var HyphaDomElement $commandsAtEnd */
+		$commandsAtEnd = $this->html->find('#pageEndCommands');
+		$commandsAtEnd->append($this->makeActionButton(__('art-cancel')));
+		$commandsAtEnd->append($this->makeActionButton(__('art-save'), self::PATH_EDIT, self::CMD_SAVE));
 	}
 
-	public function editAction() {
-		// check if form is posted and get form data
-		$formPosted = $this->isPosted(self::FORM_CMD_EDIT);
-		if ($formPosted) {
-			$formData = $_POST;
-		} else {
-			$formData = [
-				self::FIELD_NAME_TITLE => html_entity_decode(strip_tags($this->getDocPageByName(self::FIELD_NAME_TITLE)->getText())),
-				self::FIELD_NAME_AUTHOR => $this->getAttr(self::FIELD_NAME_CONTEXT, self::FIELD_NAME_AUTHOR),
-				self::FIELD_NAME_TEXT => $this->getDocPageByName(self::FIELD_NAME_TEXT)->getHtml(),
-				self::FIELD_NAME_EXCERPT => $this->getDocPageByName(self::FIELD_NAME_EXCERPT)->getHtml(),
-				self::FIELD_NAME_METHOD => $this->getDocPageByName(self::FIELD_NAME_METHOD)->getHtml(),
-				self::FIELD_NAME_SOURCES => $this->getDocPageByName(self::FIELD_NAME_SOURCES)->getHtml(),
-			];
+	/**
+	 * Changes the status of the article
+	 *
+	 * @param HyphaRequest $request
+	 * @return array
+	 */
+	public function statusChangeAction(HyphaRequest $request) {
+		if (!isUser()) {
+			notify('error', __('login-to-perform-action'));
+			return ['redirect', $this->constructFullPath($this->pagename)];
 		}
 
-		// create form
-		$form = $this->createEditForm($formData);
-
-		// process form if it was posted
-		if ($formPosted && empty($form->errors)) {
-			$this->lockAndReload();
-			$author = $form->dataFor(self::FIELD_NAME_AUTHOR);
-			$this->getDocPageByName(self::FIELD_NAME_CONTEXT)->setAttr(self::FIELD_NAME_AUTHOR, $author);
-			$title = $this->getDocPageByName(self::FIELD_NAME_TITLE);
-			$title->setText($form->dataFor(self::FIELD_NAME_TITLE));
-			$text = $this->getDocPageByName(self::FIELD_NAME_TEXT);
-			$text->setHtml($form->dataFor(self::FIELD_NAME_TEXT), true);
-			$excerpt = $this->getDocPageByName(self::FIELD_NAME_EXCERPT);
-			$excerpt->setHtml($form->dataFor(self::FIELD_NAME_EXCERPT), true);
-			$method = $this->getDocPageByName(self::FIELD_NAME_METHOD);
-			$method->setHtml($form->dataFor(self::FIELD_NAME_METHOD), true);
-			$sources = $this->getDocPageByName(self::FIELD_NAME_SOURCES);
-			$sources->setHtml($form->dataFor(self::FIELD_NAME_SOURCES), true);
-
-			$this->saveAndUnlock();
+		// TODO [LRM]: not sure that fetching the "new status" from "$_POST['argument'] is the nicest solution.
+		$newStatus = $request->getPostValue('argument');
```

Comment:
Maybe a `$request->getArgument()` would be better? That emphasizes the special nature of this `argument` argument.

Reply:
made improvements